### PR TITLE
[pkg/stanza] Always recombine if possible, even if incomplete

### PR DESCRIPTION
**Pull request by djaglowski**
Previously, certain circumstances could result in partial logs being emitted without any recombiniation. This could occur when using `is_first_entry`, if the first partial log from a source was emitted before a matching "start of log" indicator was found. This could also occur when the collector was shutting down.

### CODE REVIEW

File: pkg/stanza/operator/transformer/recombine/recombine_test.go
Comment (djaglowski) [inline pkg/stanza/operator/transformer/recombine/recombine_test.go:197]: Most of these changes are just to clarify the test. (Use meaningful content instead of "file1", etc). The primary change is that the last two tokens are emitted as one after the timeout, rather than each being emitted immediately. A complementary change below adds `ExpectEntries` because the final token from file1 and the other tokens are not emitted in a deterministic order.

**Comment by djaglowski** [comment]
@open-telemetry/collector-contrib-approvers, this PR is just waiting for a review if anyone has a moment.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**
